### ISSUE TITLE
Missing Compact/Indent/HTMLEscape comparing to encoding/json

### ISSUE THREAD

**Issue by choleraehyq** [json-iterator/go#229]
Is there any plan to support these functions?

**Comment by taowen** [comment]
no one is asking for them

**Comment by veqryn** [comment]
HTML Escaping is supported through the following, I think:
```go
func jsonIteratorMarshalAddNewLineNoEscapeHTML(v interface{}) ([]byte, error) {
	buffer := &bytes.Buffer{}
	encoder := jsonIterator.NewEncoder(buffer)
	encoder.SetEscapeHTML(false)
	err := encoder.Encode(v)
	return buffer.Bytes(), err
}
```

**Comment by carbocation** [comment]
The readme says "100% compatibility with standard lib". I think that if this line were removed, this issue could be resolved...

**Comment by shatil** [comment]
#313 relates to this. `jsoniter` doesn't support the `Umarshaler` interface: https://golang.org/pkg/encoding/json/#Unmarshaler

**Comment by jszwedko** [comment]
I agree that the README should be updated to convey which parts are not compatible with `encoding/json` and not say "100% compatible". There are a couple functions missing from the `Decoder` type:

```
        func (dec *Decoder) InputOffset() int64
        func (dec *Decoder) Token() (Token, error)
```

I had wanted to switch to `json-iterator/go`, but our code currently depends on these functions for one of our packages that does stream processing of documents and I was surprised to find them missing given the claim of drop-in compatibility.

**Comment by Dentrax** [comment]
> no one is asking for them

Any updates on this? I'm asking. 👋 I'm just hit this issue. 🙈

**Comment by damnever** [comment]
Another issue: https://github.com/json-iterator/go/issues/657

**Comment by CHN-STUDENT** [comment]
Hey，how can i export '&' char on my json by using json-iterator,could every one give me advice?
